lib/task/import/validate/csvLegacyIdValidator.class.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Access to Memory (AtoM) software.
+ *
+ * Access to Memory (AtoM) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Access to Memory (AtoM) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * CSV legacyId test. Check if legacyId is blank.
+ * Output error status and any rows where legacyId is not found.
+ *
+ * @author     Steve Breker <sbreker@artefactual.com>
+ *
+ * @internal
+ * @coversNothing
+ */
+class CsvLegacyIdValidator extends CsvBaseValidator
+{
+    const TITLE = 'LegacyId check';
+    // Persist across multiple CSVs.
+    protected $legacyIdValues = [];
+
+    // Reset after every CSV.
+    protected $legacyIdColumnPresent;
+    protected $rowsWithoutLegacyId = 0;
+    protected $nonUniqueLegacyIdValues = [];
+
+    public function __construct(array $options = null)
+    {
+        $this->setTitle(self::TITLE);
+
+        parent::__construct($options);
+    }
+
+    public function reset()
+    {
+        $this->legacyIdColumnPresent = null;
+        $this->rowsWithoutLegacyId = 0;
+        $this->nonUniqueLegacyIdValues = [];
+
+        parent::reset();
+    }
+
+    public function testRow(array $header, array $row)
+    {
+        parent::testRow($header, $row);
+        $row = $this->combineRow($header, $row);
+
+        // Is legacyId column present?
+        if (!isset($this->legacyIdColumnPresent)) {
+            $this->legacyIdColumnPresent = array_key_exists('legacyId', $row);
+        }
+
+        if ($this->legacyIdColumnPresent) {
+            if (empty($row['legacyId'])) {
+                ++$this->rowsWithoutLegacyId;
+                $this->testData->addDetail(implode(',', $row));
+            } else {
+                if (in_array($row['legacyId'], $this->legacyIdValues)) {
+                    // This is a duplicate legacyId. Add to list of non-unique legacyIds.
+                    $this->nonUniqueLegacyIdValues[] = $row['legacyId'];
```

Comment:
This will add multiple instances of `$row['legacyId']` to `$this->nonUniqueLegacyIdValues[]` if the same legacy id is used more than twice.

Reply:
Definitely unintentional. 🦅 👁️

Reply:
Fixed and updated existing unit test for this condition.